### ISSUE TITLE
Webview crash when javascript is blocked globally 

### ISSUE THREAD

**Issue by srirambv** [brave/brave-browser#174]
## Description 
Webview crash when javascript is blocked globally 

## Steps to Reproduce

   1. Enable block javascript globally 
   2. Visit brianbondy.com 
   3. Enable `Always allow Javascript` in URL bar 
   4. Webview crashes in a minute

 ## Actual result:
![image](https://user-images.githubusercontent.com/17010094/39304359-6d2429ae-4927-11e8-9597-adf61aa8a524.png)

Console logs the error
```
$ npm run start

> brave@1.0.0 start C:\brave-browser
> node ./scripts/commands.js start

C:\brave-browser\src\out\Release\brave.exe --enable-logging --v=0
[14460:25116:0426/073539.145:ERROR:dat_file_util.cc(40)] Initialized brave shields service correctly
[14460:13648:0426/073540.207:ERROR:dat_file_util.cc(40)] Initialized brave shields service correctly
[14460:252:0426/073553.341:ERROR:navigation_entry_screenshot_manager.cc(135)] Invalid entry with unique id: 8
Received fatal exception 0xc0000008
Backtrace:
        KiRaiseUserExceptionDispatcher [0x00007FF9E2F3406A+3a]
        CancelIo [0x00007FF9DFE5BC4F+f]
        mojo::edk::Channel::Create [0x00007FF97A92949F+16cf] (C:\brave-browser\src\mojo\edk\system\channel_win.cc:c9)
        base::internal::RetainedRefWrapper<base::RefCountedString>::RetainedRefWrapper<base::RefCountedString> [0x00007FF978E9A443+ef3] (C:\brave-browser\src\base\bind_internal.h:d3)
        base::internal::RetainedRefWrapper<base::RefCountedString>::RetainedRefWrapper<base::RefCountedString> [0x00007FF978E9A3E0+e90] (C:\brave-browser\src\base\bind_internal.h:126)
        base::internal::RetainedRefWrapper<base::RefCountedString>::RetainedRefWrapper<base::RefCountedString> [0x00007FF978E9A36B+e1b] (C:\brave-browser\src\base\bind_internal.h:170)
        base::internal::RetainedRefWrapper<base::RefCountedString>::RetainedRefWrapper<base::RefCountedString> [0x00007FF978E9A28D+d3d] (C:\brave-browser\src\base\bind_internal.h:15e)
        base::debug::TaskAnnotator::RunTask [0x00007FF9790D0C4A+da] (C:\brave-browser\src\base\debug\task_annotator.cc:35)
        base::MessageLoop::RunTask [0x00007FF9790B116C+23c] (C:\brave-browser\src\base\message_loop\message_loop.cc:18f)
        base::MessageLoop::DoWork [0x00007FF9790B16EA+1aa] (C:\brave-browser\src\base\message_loop\message_loop.cc:1ce)
        base::MessagePumpForIO::DoRunLoop [0x00007FF979105211+141] (C:\brave-browser\src\base\message_loop\message_pump_win.cc:1db)
        base::MessagePumpWin::Run [0x00007FF9791045D8+68] (C:\brave-browser\src\base\message_loop\message_pump_win.cc:3a)
        base::RunLoop::Run [0x00007FF97909BE05+35] (C:\brave-browser\src\base\run_loop.cc:88)
        base::Thread::ThreadMain [0x00007FF9790BE8D0+180] (C:\brave-browser\src\base\threading\thread.cc:152)
        base::PlatformThread::SetCurrentThreadPriority [0x00007FF97908A7B5+135] (C:\brave-browser\src\base\threading\platform_thread_win.cc:5e)
        BaseThreadInitThunk [0x00007FF9E1231FE4+14]
        RtlUserThreadStart [0x00007FF9E2EFF061+21]
```

## Expected result:
Should not crash webview 

## Reproduces how often: 

Easy

## Brave version (about:brave info)

N/A

### Reproducible on current release:

N/A

### Website problems only:  
- Does the issue gets fixed with Brave Shields disabled? 
- Does the issue reproduces on Chrome? 

### Additional Information
cc: @bbondy @bridiver

**Comment by srirambv** [comment]
Console log for the steps mentioned but doesn't cause webview crash. This is tested on  aa02380
```
Backtrace:
        KiRaiseUserExceptionDispatcher [0x00007FFD8663DD7A+3a]
        CancelIo [0x00007FFD82C01E1F+f]
        mojo::edk::`anonymous namespace'::ChannelWin::ShutDownOnIOThread [0x00007FFD0B30183F+ef] (C:\brave-browser\src\mojo\edk\system\channel_win.cc:9f)
        base::internal::FunctorTraits<void (mojo::edk::(anonymous namespace)::ChannelWin::*)(),void>::Invoke<scoped_refptr<mojo::edk::(anonymous namespace)::ChannelWin>> [0x00007FFD0B3011C3+43] (C:\brave-browser\src\base\bind_internal.h:1bf)
        base::internal::InvokeHelper<0,void>::MakeItSo<void (mojo::edk::(anonymous namespace)::ChannelWin::*)(),scoped_refptr<mojo::edk::(anonymous namespace)::ChannelWin> > [0x00007FFD0B301166+56] (C:\brave-browser\src\base\bind_internal.h:212)
        base::internal::Invoker<base::internal::BindState<void (mojo::edk::(anonymous namespace)::ChannelWin::*)(),scoped_refptr<mojo::edk::(anonymous namespace)::ChannelWin> >,void ()>::RunImpl<void (mojo::edk::(anonymous namespace)::ChannelWin::*)(),std::tuple< [0x00007FFD0B301109+49] (C:\brave-browser\src\base\bind_internal.h:25c)
        base::internal::Invoker<base::internal::BindState<void (mojo::edk::(anonymous namespace)::ChannelWin::*)(),scoped_refptr<mojo::edk::(anonymous namespace)::ChannelWin> >,void ()>::RunOnce [0x00007FFD0B301046+46] (C:\brave-browser\src\base\bind_internal.h:23c)
        base::debug::TaskAnnotator::RunTask [0x00007FFD0CC2B455+135] (C:\brave-browser\src\base\debug\task_annotator.cc:65)
        base::MessageLoop::RunTask [0x00007FFD0CC0585C+23c] (C:\brave-browser\src\base\message_loop\message_loop.cc:163)
        base::MessageLoop::DoWork [0x00007FFD0CC05DB8+198] (C:\brave-browser\src\base\message_loop\message_loop.cc:198)
        base::MessagePumpForIO::DoRunLoop [0x00007FFD0CC1AABA+14a] (C:\brave-browser\src\base\message_loop\message_pump_win.cc:1df)
        base::MessagePumpWin::Run [0x00007FFD0CC19DB8+68] (C:\brave-browser\src\base\message_loop\message_pump_win.cc:3b)
        base::RunLoop::Run [0x00007FFD0CBEF241+31] (C:\brave-browser\src\base\run_loop.cc:88)
        base::Thread::ThreadMain [0x00007FFD0CC156D0+180] (C:\brave-browser\src\base\threading\thread.cc:154)
        base::`anonymous namespace'::ThreadFunc [0x00007FFD0CBDD074+f4] (C:\brave-browser\src\base\threading\platform_thread_win.cc:5e)
        BaseThreadInitThunk [0x00007FFD85BF3034+14]
        RtlUserThreadStart [0x00007FFD86611551+21]
```

**Comment by bbondy** [comment]
@srirambv Does this reproduce in release builds or only debug? Still valid?

**Comment by srirambv** [comment]
This looks fixed on  c84b457. No webview crash based on the steps

**Comment by GeetaSarvadnya** [comment]
Verified on:

Brave | 0.54.2 Chromium: 69.0.3497.42 (Official Build) dev (64-bit)
-- | --
Revision | 9c4613c66dfeb2b76ef6dd4b15884c15db3b4969-refs/branch-heads/3497@{#655}
OS | Windows
JavaScript | V8 6.9.427.14
Flash | 30.0.0.154
